adafruit_esp32spi/adafruit_esp32spi_socket.py

```diff
@@ -161,16 +161,20 @@ def recv(self, bufsize=0):
         gc.collect()
         return ret
 
-    def recv_into(self, buffer):
+    def recv_into(self, buffer, nbytes=None):
         """Read some bytes from the connected remote address into a given buffer
 
         :param bytearray buffer: The buffer to read into
+        :param int nbytes: (Optional) Number of bytes to receive
+            default is as many as possible before filling the
+            buffer or timing out
         """
 
         stamp = time.monotonic()
         to_read = len(buffer)
+        limit = 0 if nbytes is None else to_read - nbytes
         received = []
-        while to_read > 0:
+        while to_read > limit:
```

Comment:
I think this while loop is evaluating to false and we aren't going inside of it. I added a debugging print above this to print the values of limit and to_read: `print("to_read: {} - limit: {}".format(to_read, limit))`

And I'm seeing these:
```
to_read: 32 - limit: 32
```

I also tried modifying the requests library to not pass the `nbytes` argument and it seemed to get stuck in an infinite loop, but did eventually finish after a few minutes. I think it took much longer than I was expecting based on previous experience running the simpletest. But I'm not sure if this part is actually related to this issue since requests does pass `nbytes` and that is how this is intended to work as far as I understand.